### PR TITLE
[main] Update kernel to v5.15.34.1 to address several CVEs

### PR DESCRIPTION
**Pull request by Camelron**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Update kernel to v5.15.34.1 to address several CVEs

###### Change Log  

- Update kernel source to 5.15.34.1
- Clean up nopatch entries in kernel.spec, no longer needed by our tooling
- nopatch CVE-2022-28390, CVE-2022-28389, CVE-2022-28388, CVE-2022-28356, CVE-2022-0435,
    CVE-2021-4202, CVE-2022-27950, CVE-2022-0433, CVE-2022-0494, CVE-2022-0330, CVE-2022-0854, 
    CVE-2021-4197

###### Does this affect the toolchain?  

YES

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2022-28390
- https://nvd.nist.gov/vuln/detail/CVE-2022-28389
- https://nvd.nist.gov/vuln/detail/CVE-2022-28388
- https://nvd.nist.gov/vuln/detail/CVE-2022-28356
- https://nvd.nist.gov/vuln/detail/CVE-2022-0435
- https://nvd.nist.gov/vuln/detail/CVE-2021-4202
- https://nvd.nist.gov/vuln/detail/CVE-2022-27950
- https://nvd.nist.gov/vuln/detail/CVE-2022-0433
- https://nvd.nist.gov/vuln/detail/CVE-2022-0494
- https://nvd.nist.gov/vuln/detail/CVE-2022-0330
- https://nvd.nist.gov/vuln/detail/CVE-2022-0854
- https://nvd.nist.gov/vuln/detail/CVE-2021-4197

###### Test Methodology

- AMD64: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=186583&view=logs&j=75cba72d-09cb-503a-3927-4ab61ac4a179&t=266a177c-c74b-572c-2e05-3a6c5cd5b38a (ruby package failed, but kernel built)
- ARM64: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=186700&view=logs&j=572904f9-79b1-51c4-99c0-531b4ffd4297&t=5cec731e-43d6-5f41-37bc-70613258898a (ruby package failed, but kernel built)

### CODE REVIEW

**Review by neha170 — APPROVED** [review]
Thanks for adding the script to update kernel. Could you add an example input to the script, thanks

File: toolkit/scripts/update_kernel.sh [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    echo sha256 = $SHA256
+}
+
+
+function usage() { 
+    echo "Update sources for kernel" 
+    echo "v : Version you are updating to (ex. 5.10.37.1)" 
+    echo "u : Your name"
+    echo "e : Your email"
+    echo "w : Absoulte patch to your workspace for your update - no quotes"
```
Comment (neha170) [inline toolkit/scripts/update_kernel.sh:192]: Did you mean path?

File: toolkit/scripts/update_kernel.sh [hunk 2]
```diff
@@ -0,0 +1,275 @@
+#!/bin/bash
```
Comment (christopherco) [inline toolkit/scripts/update_kernel.sh:1]: Add standard MS Copyright header
```suggestion
#!/bin/bash
# Copyright (c) Microsoft Corporation.
# Licensed under the MIT License.
```

**Review by christopherco — APPROVED** [review]

File: toolkit/scripts/update_kernel.sh (hunk 1 above)
Comment (christopherco) [reply toolkit/scripts/update_kernel.sh:192]: nit - Also `s/Absoulte/Absolute`
```suggestion
    echo "w : Absolute path to your workspace for your update - no quotes"
```

**Pull request merged**